content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1007 @@
+let (a, query) = system_state.get(&world);
+```
+
+{{rust_type(type="struct" crate="bevy_ecs" version="0.6.0" name="SystemState")}} does all of the same caching that a normal Bevy system does, so reusing the same SystemState results in uber-fast World access.
+
+### Sub Apps
+
+<div class="release-feature-authors">authors: @cart, @zicklag, @bjorn3</div>
+
+The new Bevy renderer requires strict separation between the "main app" and the "render app". To enable this, we added the concept of "sub apps":
+
+```rust
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, AppLabel)]
+pub struct RenderApp;
+
+let mut render_app = App::empty();
+app.add_sub_app(RenderApp, render_app, move |app_world, render_app| {
+    // execute app logic here
+});
+
+// later
+app.sub_app_mut(RenderApp)
+    .add_system(some_system);
+    .add_system(some_other_system);
+```
+
+We plan on exposing more control over scheduling and running sub apps in the future.
+
+### Query::iter_combinations
+
+<div class="release-feature-authors">authors: @Frizi</div>
+
+You can now iterate all permutations of N entities for a given query:
+
+```rust
+fn system(query: Query<&Player>) {
+    // iterates each possible combination of two entities exactly once
+    for [e1, e2] in query.iter_combinations() {
+    }
+
+    // iterates each possible combination of three entities exactly once
+    for [e1, e2, e3] in query.iter_combinations() {
+    }
+}
+```
+
+This is especially useful for things like "checking for entities for collisions with all other entities". There is also an `iter_combinations_mut` variant. Just be careful ... the time complexity of this grows quickly with large numbers of combinations. With great power comes great responsibility!
+
+The new `iter_combinations.rs` example illustrats how to use this new API to calculate gravity between objects in a "solar system":
+
+![iter_combinations](iter_combinations.png)
+
+### Optimized System Commands
+
+<div class="release-feature-authors">authors: @NathanSWard</div>
+
+System Commands got a nice performance boost by changing how command buffers are stored and reused:
+
+#### Entity Spawn Benchmark Duration (in microseconds, less is better)
+
+This benchmark spawns entities with a variety of component compositions to ensure we cover a variety of cases. Treat these numbers as realtives, not absolutes.
+
+![commands perf](commands_perf.svg)
+
+### System Param Lifetimes
+
+<div class="release-feature-authors">authors: @cart, @BoxyUwU, @TheRawMeatball</div>
+
+System and Query lifetimes were made more explicit by splitting out the `'system` and `'world`, lifetimes and using them explicitly where possible. This enables Rust to reason about ECS lifetimes more effectively, especially for read-only lifetimes. This was particularly important because it enabled the new Bevy Renderer to convince WGPU that ECS resources actually live for as long as the Render World.
```

Comment:
```suggestion
System and Query lifetimes were made more explicit by splitting out the `'system` and `'world`, lifetimes and using them explicitly where possible. This enables Rust to reason about ECS lifetimes more effectively, especially for read-only lifetimes. This was particularly important because it enabled the new Bevy Renderer to convince wgpu that ECS resources actually live for as long as the Render World.
```